cirq/circuits/circuit.py

```diff
@@ -666,16 +666,41 @@ def findall_operations_until_blocked(
             is_blocker: Callable[[ops.Operation], bool] = lambda op: False
     ) -> List[Tuple[int, ops.Operation]]:
         """
-        Finds all operations until a blocking operation is hit.  This returns
-        a list of all operations from the starting frontier until a blocking
-        operation is encountered.  An operation is part of the list if
-        it is involves a qubit in the start_frontier dictionary, comes after
-        the moment listed in that dictionary, and before any blocking
-        operations that involve that qubit.  Operations are only considered
-        to be blocking the qubits that they operate on, so a blocking operation
-        that does not operate on any qubit in the starting frontier is not
-        actually considered blocking.  See `reachable_frontier_from` for a more
-        in depth example of reachable states.
+        Finds all operations until a blocking operation is hit.
+
+        An operation is considered blocking if
+
+        a) It is in the 'light cone' of start_frontier.
```

Comment:
What does 'light cone' of start_frontier mean?

I think these semantics are pretty weird, but maybe I am not understanding it correctly.

Reply:
Basically, if you create the circuit DAG (using the default `can_reorder`), I think of the "light cone" of an operation as the successors of that node in the DAG. The light cone of a position (qubit, moment_index) is the set of positions at which operations would be in the light cone of an operation at the first position.

Reply:
With the current implementation (in master), for

the circuit
```
0: ───ZZ────────────
      │
1: ───ZZ───ZZ───Z───
           │
2: ───XX───ZZ───────
      │
3: ───XX────────────
```

`start_frontier[q] = 0` for all four qubits and `is_blocker = lambda op: isinstance(op.gate, cirq.XXPowGate)`, 

`findall_all_operations_until_blocked` returns all the Z/ZZ gates.

That seems surprising, because all but the first one "should" be blocked by the XX.